source/PlayState.hx

```diff
@@ -1,21 +1,63 @@
 package;
 
 import actors.player.Hero;
+import environment.Door;
+import environment.TotalKeys;
+import environment.Key;
+import flixel.group.FlxGroup.FlxTypedGroup;
 import flixel.FlxState;
+import flixel.FlxG;
+import flixel.FlxObject;
 
 class PlayState extends FlxState
 {
 	private var hero:Hero;
+	private var doors:FlxTypedGroup<Door>;
+	private var door:Door;
+	private var keys:FlxTypedGroup<Key>;
+	private var key:Key;
+	private var totalKeys:TotalKeys;
 
 	override public function create():Void
 	{
 		super.create();
 		hero = new Hero();
 		add(hero);
+
+		doors = new FlxTypedGroup<Door>();
+		door = new Door(50, 50);
+		doors.add(door);
+		door = new Door(250, 250);
+		doors.add(door);
+		add(doors);
+
+		keys = new FlxTypedGroup<Key>();
+		key = new Key(100, 100);
+		keys.add(key);
+		key = new Key(300, 300);
+		keys.add(key);
+		add(keys);
+
+		totalKeys = new TotalKeys();
 	}
 
 	override public function update(elapsed:Float):Void
 	{
+		if(FlxObject.updateTouchingFlags(hero, doors))
+			{
+				if(totalKeys.getKey() > 1)
+					{
+						totalKeys.useKeys();
+						doors.open();
+					}
+			}
+		if(FlxObject.updateTouchingFlags(hero, keys))
+			{
+				totalKeys.pickup();
+				keys.pickup();
+			}
+		FlxG.collide(hero, doors);
+		
```

Comment:
The weird behavior with getting multiple keys is happening because of the FlxObject.updateTouchingFlags, I think.

For this kind of logic, we want to use the [FlxG.overlap() method](https://api.haxeflixel.com/flixel/FlxG.html#overlap) instead

```
FlxG.overlap(hero, keys, resolveHeroAndKeyOverlap);
FlxG.overlap(hero, doors, resolveHeroAndDoorOverlap);
```
Where `resolveHeroAndKeyOverlap` and `resolveHeroAndDoorOverlap` are methods that take a hero object and the other type of object and do the appropriate logic on them.

This will work similarly to the `itsAHit()` method in our example code from class: https://gist.github.com/SamBumgardner/dced222851a2e77dfaf5e743165a0bb0#file-playstate-hx-L55

Where this overlap-resolution function should live is a bit of a trickier problem, since it involves multiple objects. Keeping it in the PlayState seems fine to me for now, but it's something we can discuss in more detail later.